File: software/examples/rust/example_morse_code.rs
use std::{io, error::Error};

use tinkerforge::{ip_connection::IpConnection, 
                  piezo_speaker_bricklet::*};


const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Piezo Speaker Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let ps = PiezoSpeakerBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
    // Don't use device before ipcon is connected.

		// Morse SOS with a frequency of 2kHz
		ps.morse_code("... --- ...".to_string(), 2000);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}

File: software/examples/go/example_morse_code.go
package main

import (
	"fmt"
	"github.com/tinkerforge/go-api-bindings/ipconnection"
	"github.com/tinkerforge/go-api-bindings/piezo_speaker_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Piezo Speaker Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	ps, _ := piezo_speaker_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Morse SOS with a frequency of 2kHz
	ps.MorseCode("... --- ...", 2000)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}
